File: Assets/Scripts/Nivel1/enemigo Final/JEFE.js
#pragma strict


var iniciando:boolean=true;

var velocidad:float=3.0f;


var generadorColmillos1:GameObject;
var generadorColmillos2:GameObject;
var generadorColmillos3:GameObject;
var generadorColmillos4:GameObject;
var generadorColmillos5:GameObject;
var generadorColmillos6:GameObject;

var colmillos;
function Start () {
	colmillos=[generadorColmillos1,generadorColmillos2,generadorColmillos3,generadorColmillos4,generadorColmillos5,generadorColmillos6];
}

function Update () {
	if(iniciando)
		avanzar();

}


function avanzar(){
	this.transform.Translate(Vector3.left*velocidad*Time.deltaTime);

	if(this.transform.position.x<=0){
		iniciando=false;

		for(var i=0; i<6;i++){
			//colmillos[i].gameObject.GetComponent.<generadorColmillos>().enPosicion();
		}

	}
}

File: Assets/Scripts/Nivel1/enemigo Final/JEFE.cs
using UnityEngine;
using System.Collections;
using UnityEngine.UI;

public class JEFE : MonoBehaviour {


	bool iniciando=false;
	bool cambiandoMusica=true;

	public float velocidad=3.0f;



	public AudioSource musica;
	public AudioClip cancionJefe;

	public Slider vidaJefe;

	// Use this for initialization
	void Start () {
		
	}
	
	// Update is called once per frame
	void Update () {
		if(iniciando)
			avanzar();
		if(cambiandoMusica){
			cambiarMusica();
		}

	}



	void avanzar(){
		this.transform.Translate(Vector3.left*velocidad*Time.deltaTime);




		if(this.transform.position.x<=0){
			iniciando=false;


			vidaJefe.gameObject.SetActive(true);
		}
	}



	void cambiarMusica(){
		musica.volume-=0.2f*Time.deltaTime;
		if(musica.volume<=0){
			musica.Stop();
			musica.clip=cancionJefe;
			musica.volume=1.0f;
			cambiandoMusica=false;
			iniciando=true;
			musica.Play();
		}
	}



}
